### PR TITLE
(RFC) Csharp netcore generator supports UnityWebRequest library

### PR DESCRIPTION
**Pull request by justonia**
This PR is a working implementation of a library plugin for the csharp-netcore client generator that utilizes `UnityWebRequest` as the request backend.

Why, when RestSharp and HttpClient work in Unity? Cross-platform nightmares -- especially consoles.

Limitations (as specific as I can be due to NDA):

* Console A - Anything except UnityWebRequest doesn't support access to the platform's SSL certificate store. You have to embed certificates at build time and apply for a special waiver to use them. This then becomes a ticking time bomb because your root certs can expire while your game is in the wild.
* Console B - With no exception, requires you to use UnityWebRequest since it integrates natively with the platform's custom version of libcurl that all games must use.
* Console C - Use whatever you want! Villagers rejoice.
* File parameters cannot be supported everywhere. Some consoles and tvOS do not support either `Application.persistentDataPath` or will refuse to compile/link if you even reference `GetTempPath` as well.
* Setting cookies are not supported in WebGL platform
* UnityWebRequest also offers native performance with much less garbage usage. It's worth noting that the current Unity GC is still a very old one and even with incremental GC support it's still problematic on min-spec devices.

I am genuinely not sure how to proceed with this as a PR that could be considered for integration. First off, unit tests would require a Unity project, and secondly, I'm sure some of the changes I've made might be controversial.

The backend I've written also adds better result handling and provides additional exceptions that let you know when an API call failure is due to a connection error. The latter is a must-have distinction because console cert processes will fail you if your error messages are not exactly specific as to what happened (e.g. no "something went wrong" when you have no internet).

This backend only has one C# dependency that isn't default in Unity (or available via package manager like JSON.net): `System.ComponentModel.DataAnnotations.dll`.

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CSharpNetCoreClientCodegen.java [hunk 1]
```diff
@@ -99,6 +100,7 @@ public class CSharpNetCoreClientCodegen extends AbstractCSharpCodegen {
 
     protected boolean supportsRetry = Boolean.TRUE;
     protected boolean supportsAsync = Boolean.TRUE;
+    protected boolean supportsFileParameters = Boolean.TRUE;
```
Comment (justonia) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CSharpNetCoreClientCodegen.java:103]: I did not expose this as a configuration option as it did not seem appropriate to expose for other backends.

**Comment by wing328** [comment]
is https://github.com/Sunblink-Entertainment/openapi-generator-unity.git a private repo? can you make it public?

I can't pull the changes to test it locally via
```
git checkout -b Sunblink-Entertainment-unity-webrequest master
git pull https://github.com/Sunblink-Entertainment/openapi-generator-unity.git unity-webrequest
```
(commands provided by github)

**Comment by justonia** [comment]
@wing328 Ah, this got messed up on our end. I will fork and reopen a new PR with this content.

**Comment by pixelpax** [comment]
@justonia did this ever get resolved? Is there a working solution now ?

**Comment by justonia** [comment]
Closing because  PR #14870 is taking the reins.

**Pull request closed without merging**